components/core/src/env.rs

```diff
@@ -248,6 +248,7 @@ macro_rules! env_config_string {
 macro_rules! default_as_str {
     ($wrapping_type:ident) => {
         impl $wrapping_type {
+            #[allow(dead_code)] // I'm not clear why, but this is required to avoid a warning on default_as_str()
```

Comment:
It's because [this invocation](https://github.com/habitat-sh/habitat/blob/master/components/core/src/util/sys.rs#L14-L21) of `env_config_socketaddr` doesn't use `default_as_str`, whereas all the others do, because they need to provide defaults to CLAP and this instance does not.

Reply:
Ahh, thank you